plugin/core/rpc.py

```diff
@@ -191,9 +191,12 @@ def execute_request(
     def flush_deferred_notifications(self) -> None:
         for payload in self._deferred_notifications:
             try:
-                handler = self._get_handler(payload["method"])
+                method = payload["method"]
+                handler = self._get_handler(method)
+                result = payload["params"]
+                self._logger.incoming_notification(method, result, handler is None)
```

Comment:
Surprising that this wasn't noticed earlier :)

Reply:
There is also `Client.response_handler` where we, in some cases, construct our own error response and the original payload is not logged anywhere. That also makes debugging harder.